test-network-function/accesscontrol/suite.go

```diff
@@ -125,26 +141,47 @@ func runTestOnPods(env *config.TestEnvironment, testCmd testcases.BaseTestCase,
 			if count > 0 {
 				count := 0
 				for count < podUnderTest.ContainerCount {
+					ginkgo.By(fmt.Sprintf("Executing TC %s on pod %s (ns %s), container index %d", testCmd.Name, podNamespace, podName, count))
 					argsCount := append(args, count)
-					cmdArgs := strings.Split(fmt.Sprintf(testCmd.Command, argsCount...), " ")
+					cmd := fmt.Sprintf(testCmd.Command, argsCount...)
+					cmdArgs := strings.Split(cmd, " ")
 					cnfInTest := containerpkg.NewPod(cmdArgs, podUnderTest.Name, podUnderTest.Namespace, testCmd.ExpectedStatus, testCmd.ResultType, testCmd.Action, common.DefaultTimeout)
 					gomega.Expect(cnfInTest).ToNot(gomega.BeNil())
 					test, err := tnf.NewTest(context.GetExpecter(), cnfInTest, []reel.Handler{cnfInTest}, context.GetErrorChannel())
 					gomega.Expect(err).To(gomega.BeNil())
 					gomega.Expect(test).ToNot(gomega.BeNil())
-					test.RunAndValidate()
+					test.RunWithCallbacks(nil, func() {
+						tnf.ClaimFilePrintf("Command sent: %s, Expectations: %v", cmd, testCmd.ExpectedStatus)
+						addFailedTcInfo(failedTcs, testCmd.Name, podName, podNamespace, count)
+					}, func(e error) {
+						tnf.ClaimFilePrintf("Command sent: %s, Expectations: %v, Error: %v", cmd, testCmd.ExpectedStatus, e)
```

Comment:
Prepend the log with the error type to improve readability, for instance:
TC failed (ERROR): Command sent: %s, Expectations: %v, Error: %v", c
TC failed (FAILURE): Command sent: %s, Expectations: %v", c